model/status_diff.go

```diff
@@ -54,45 +55,60 @@ var (
 
 // StatusDiffBuilds takes two builds and returns a diff of their results
 // for easy comparison and analysis.
-func StatusDiffBuilds(original, patch *build.Build) BuildStatusDiff {
+func StatusDiffBuilds(original, patch *build.Build) (BuildStatusDiff, error) {
 	// return an empty diff if one of builds is nonexistant
 	// this is likely to occur after adding a new buildvariant or task
 	if original == nil || patch == nil {
-		return BuildStatusDiff{}
+		return BuildStatusDiff{}, nil
 	}
 
 	diff := BuildStatusDiff{
 		Name: original.DisplayName,
 		Diff: StatusDiff{original.Status, patch.Status},
 	}
 
-	// build maps of task statuses, for matching
-	originalTasks := make(map[string]build.TaskCache)
-	for _, task := range original.Tasks {
-		originalTasks[task.DisplayName] = task
+	tasks, err := task.FindAll(
+		task.ByBuildIds([]string{original.Id, patch.Id}).WithFields(task.BuildIdKey, task.StatusKey, task.DetailsKey, task.DisplayNameKey))
```

Comment:
Do we have a good index for this? It could be faster otherwise to do `task.ByIds(originalAndPatchTaskIds)`

Reply:
we have an index on build_id